### PR TITLE
Hpo tutorial basic

### PR DESCRIPTION
**Pull request by annaelisalappe**
# Summary

This PR adds a second HPO tutorial on the second, simpler way of adding HPO functionalities to an existing itwinai Trainer and Pipeline. There is again a tutorial using the FashionMNIST dataset and reset18 model to try and keep the two HPO tutorials similar. Hopefully this will allow users to see the differences easily. I also extended the HPO introduction by some new sections that highlight the differences between the two methods, guide the user for when to use which integration, and introduces the tutorial. 

Note that the other tutorial (on the `RayTorchTrainer`) was moved to another folder, so those files appear here as well

### CODE REVIEW

File: docs/tutorials/hpo-workflows/hpo-basic-integration.rst [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+    def run_trial(config: Dict, data: Dict):
+        pipeline_name = data["pipeline_name"]
+        parser = ConfigParser(
+            config="config.yaml",
+            override_keys={
+                # Set hyperparameters controlled by ray
+                "batch_size": config["batch_size"],
+                "optim_lr": config["optim_lr"],
+                # Override logger field, because performance is logged by ray
+                # f"{pipeline_name}.init_args.steps.1.init_args.logger": None,
```
[the remark this answers is not in the document]
Comment (annaelisalappe) [reply docs/tutorials/hpo-workflows/hpo-basic-integration.rst:67]: Nope 🙃This was from before I found out you can just use the loggers and it's much better -  I'll remove it

File: docs/how-it-works/hpo/explain-hpo.rst [hunk 2]
```diff
@@ -110,17 +112,49 @@ How a Ray Cluster Operates:
...
 
 
-How Distributed Training Works with the RayTorchTrainer
---------------------------------------------------------
+How to Run Your Pipeline with Ray Tune
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The easiest way to start running HPO with itwinai is to use our template to wrap a 
+pipeline in a simple function to pass it to a Ray Tune ``Tuner``. This method is suitable for users who want a quick, lightweight setup -
+if you already have an itwinai trainer and pipeline, setting up this integration should not take you
+much more than ten minutes. This method uses only Ray Tune for trial distribution and hyperparameter sampling,
```
Comment (jarlsondre) [inline docs/how-it-works/hpo/explain-hpo.rst:128]: Not a big deal, but I think I would omit the direct mention of "ten minutes" here and rather say something more generic like "only includes a few short steps" or something along those lines.

File: docs/tutorials/hpo-workflows/hpo-basic-integration.rst [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+*    Add or remove hyperparameters to suit your model.
+*    Use ``tune.choice`` for discrete parameters and ``tune.uniform`` for continuous, uniform ranges. You can add different sampling distributions and types. For additional search space options, please refer to the `Ray Tune documentation <https://docs.ray.io/en/latest/tune/api/search_space.html>`_. Any search space definition supported by Ray Tune can be used in this template as well.
+
+**2.  Define ray configs:** 
+Set up the tuning and run configurations to control the number of trials, optimization goal, and stopping criteria.
+
+.. code-block:: python
+
+    tune_config = tune.TuneConfig(
+                metric=args.metric,  # Metric to optimize (loss by default)
```
Comment (jarlsondre) [inline docs/tutorials/hpo-workflows/hpo-basic-integration.rst:110]: You don't necessarily need to change anything here, but was just wondering: What is the type of `args.metric` here? Is it a callable or something?

Comment (annaelisalappe) [reply docs/tutorials/hpo-workflows/hpo-basic-integration.rst:110]: No it is just a string ("min" or "max"). In the full code you can see this in the command-line arguments 😌

**Review by jarlsondre — APPROVED** [review]
🔥

**Pull request merged**